### ISSUE TITLE
NeoVim doesn't restore default text cursor when exiting

### ISSUE THREAD

**Issue by wbpayne22902** [neovim/neovim#18946]
### Neovim version (nvim -v)

NVIM 0.7.0

### Vim (not Nvim) behaves the same?

no 8.2.5076

### Operating system/version

Windows 11 v. 10.0.21536.1000

### Terminal name/version

Windows Terminal and Windows Console Host

### $TERM environment variable

windows default

### Installation

chocolatey

### How to reproduce the issue

Open neovim, the text cursor changes to a full block cursor.
Exit neovim, the text cursor remains a full block cursor.

### Expected behavior

When exiting neovim the previous text cursor settings are restored

### Actual behavior

The text cursor remains full block
![screenshot before starting neovim](https://user-images.githubusercontent.com/107161609/173268467-ae463ed8-ebed-41bf-8223-388dbb4ddc71.png)
![after exiting](https://user-images.githubusercontent.com/107161609/173268476-77963d92-2dd8-4f8c-9180-6f5b0f102094.png)

**Comment by justinmk** [comment]
Thanks for the report. Can you try the development version? 

- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. 
- Check `:version` to confirm that you are using the latest version.

**Comment by wbpayne22902** [comment]
I tried the latest development version, and the issue is unfortunately still present.

**Comment by justinmk** [comment]
> Vim (not Nvim) behaves the same?
> no 8.2.5076

Because Vim doesn't set the cursor shape at all.

> Actual behavior
> The text cursor remains full block

Oh, I mistakenly thought this issue was about leaving the cursor in I-mode.

Nvim sets block because terminals don't provide a way to query the cursor shape. 

Workaround: https://github.com/neovim/neovim/issues/4867#issuecomment-291249173

duplicate of https://github.com/neovim/neovim/issues/4396
